Does not build with lts-8.14

I've tried building udp-conduit with lts-8.14 and it fails with the following error:


```

    Configuring udp-conduit-0.1.0.4...
    Building udp-conduit-0.1.0.4...
    Preprocessing library udp-conduit-0.1.0.4...
    [1 of 1] Compiling Conduit.UDP      ( src/Conduit/UDP.hs, .stack-work/dist/x86_64-linux-nix/Cabal-1.24.2.0/build/Conduit/UDP.o )
    
    /tmp/stack8661/udp-conduit-0.1.0.4/src/Conduit/UDP.hs:56:5: error:
        • Couldn't match type ‘a’
                         with ‘bytestring-0.10.8.1:Data.ByteString.Internal.ByteString’
          ‘a’ is a rigid type variable bound by
            the type signature for:
              udpSink :: forall (m :: * -> *) a o.
                         (MonadResource m, IOData a) =>
                         String -> PortNumber -> ConduitM a o m ()
            at src/Conduit/UDP.hs:51:8
          Expected type: ConduitM a o m ()
            Actual type: ConduitM
                           bytestring-0.10.8.1:Data.ByteString.Internal.ByteString o m ()
        • In the expression: sinkIOHandle (udpHandle host port SinkSocket)
          In an equation for ‘udpSink’:
              udpSink host port = sinkIOHandle (udpHandle host port SinkSocket)
        • Relevant bindings include
            udpSink :: String -> PortNumber -> ConduitM a o m ()
              (bound at src/Conduit/UDP.hs:55:1)
    
    /tmp/stack8661/udp-conduit-0.1.0.4/src/Conduit/UDP.hs:69:5: error:
        • Couldn't match type ‘a’
                         with ‘bytestring-0.10.8.1:Data.ByteString.Internal.ByteString’
          ‘a’ is a rigid type variable bound by
            the type signature for:
              udpSource :: forall (m :: * -> *) a i.
                           (MonadResource m, IOData a, MonoFoldable a) =>
                           String -> PortNumber -> ConduitM i a m ()
            at src/Conduit/UDP.hs:64:8
          Expected type: ConduitM i a m ()
            Actual type: ConduitM
                           i bytestring-0.10.8.1:Data.ByteString.Internal.ByteString m ()
        • In the expression:
            sourceIOHandle (udpHandle host port SourceSocket)
          In an equation for ‘udpSource’:
              udpSource host port
                = sourceIOHandle (udpHandle host port SourceSocket)
        • Relevant bindings include
            udpSource :: String -> PortNumber -> ConduitM i a m ()
              (bound at src/Conduit/UDP.hs:68:1)

```

--- a/src/Conduit/UDP.hs
+++ b/src/Conduit/UDP.hs
@@ -3,8 +3,7 @@ module Conduit.UDP
     , udpSource
     ) where
 
-import Data.IOData (IOData)
-import Data.MonoTraversable (MonoFoldable)
+import Data.ByteString (ByteString)
 import System.IO (IOMode(ReadMode, WriteMode), Handle)
 import qualified Network.Socket as Socket
 import Network.Socket
@@ -48,10 +47,10 @@ socketMode socketType =
 -- from upstream toward the destination. Must be used with
 -- 'Data.Conduit.runConduitRes', since it uses a finite resource (sockets)
 udpSink
-    :: (MonadResource m, IOData a)
+    :: (MonadResource m)
     => String -- ^ address to destination
     -> PortNumber -- ^ port number on destination
-    -> ConduitM a o m () -- ^ Can be viewed as a @'Data.Conduit.Sink' ByteString@
+    -> ConduitM ByteString o m () -- ^ Can be viewed as a @'Data.Conduit.Sink' ByteString@
 udpSink host port =
     sinkIOHandle (udpHandle host port SinkSocket)
 
@@ -61,10 +60,10 @@ udpSink host port =
 -- being sent to that interface and port. Must be used with
 -- 'Data.Conduit.runConduitRes', since it uses a finite resource (sockets)
 udpSource
-    :: (MonadResource m, IOData a, MonoFoldable a)
+    :: (MonadResource m)
     => String -- ^ address to bind to
     -> PortNumber -- ^ port number to bind to
-    -> ConduitM i a m () -- ^ Can be viewed as a @MonadIO m => 'Data.Conduit.Source' m ByteString@
+    -> ConduitM i ByteString m () -- ^ Can be viewed as a @MonadIO m => 'Data.Conduit.Source' m ByteString@
 udpSource host port =
     sourceIOHandle (udpHandle host port SourceSocket)
 

--- a/stack.yaml
+++ b/stack.yaml
@@ -1,7 +1,7 @@
 # For more information, see: https://github.com/commercialhaskell/stack/blob/release/doc/yaml_configuration.md
 
 # Specifies the GHC version and set of packages available (e.g., lts-3.5, nightly-2015-09-21, ghc-7.10.2)
-resolver: lts-7.2
+resolver: lts-10.8
 
 # Local packages, usually specified by relative directory name
 packages:

--- a/udp-conduit.cabal
+++ b/udp-conduit.cabal
@@ -20,10 +20,9 @@ library
   ghc-options:         -Wall
   exposed-modules:     Conduit.UDP
   build-depends:       base >= 4.7 && < 5
-                     , mono-traversable
-                     , chunked-data
                      , network
-                     , conduit-combinators
+                     , bytestring < 0.11
+                     , conduit-combinators >= 1.1 && < 1.2
   default-language:    Haskell2010
 
 source-repository head

